### PR TITLE
Use smaller Wasm heap and stack at compile time

### PR DESCRIPTION
**Pull request by mandarjog**
Continuation of https://github.com/istio/proxy/pull/2965

### CODE REVIEW

**Comment by mandarjog** [comment]
@bianpengyuan we need to do the same in the istio-extensions repo.

File: extensions/BUILD [hunk 1]
```diff
@@ -26,6 +26,12 @@ load(
 
 envoy_package()
 
+WASM_LINKOPTS = [
+    "--js-library external/proxy_wasm_cpp_sdk/proxy_wasm_intrinsics.js",
+    "-s TOTAL_MEMORY=8MB",
+    "-s TOTAL_STACK=5MB",
```
Comment (PiotrSikora) [inline extensions/BUILD:32]: `TOTAL_STACK=5MB` is the default value, so we might skip it.

File: extensions/BUILD [hunk 2]
```diff
@@ -26,6 +26,12 @@ load(
 
 envoy_package()
 
+WASM_LINKOPTS = [
+    "--js-library external/proxy_wasm_cpp_sdk/proxy_wasm_intrinsics.js",
+    "-s TOTAL_MEMORY=8MB",
```
Comment (PiotrSikora) [inline extensions/BUILD:31]: `TOTAL_MEMORY` was renamed to `INITIAL_MEMORY` a year ago.

File: extensions/BUILD [hunk 3]
```diff
@@ -26,6 +26,12 @@ load(
 
 envoy_package()
 
+WASM_LINKOPTS = [
+    "--js-library external/proxy_wasm_cpp_sdk/proxy_wasm_intrinsics.js",
```
Comment (PiotrSikora) [inline extensions/BUILD:30]: Is this needed here?

**Review by PiotrSikora — CHANGES_REQUESTED** [review]

File: extensions/BUILD (hunk 3 above)
Comment (bianpengyuan) [reply extensions/BUILD:30]: This is not needed.

**Review by PiotrSikora — APPROVED** [review]
Thanks!

**Comment by PiotrSikora** [comment]
@bianpengyuan PTAL

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**